src/movement/test/test_control_system.cpp

```diff
@@ -0,0 +1,94 @@
+#include <gtest/gtest.h>
+#include "ros/ros.h"
+#include "robosub/control.h"
+#include "utility/test_tools.hpp"
+#include <vector>
+#include <string>
+#include "robosub/depth_stamped.h"
+
+ros::Publisher pub;
+
+//function for extracting the depth data
+double get_depth_data(const robosub::depth_stamped::ConstPtr& msg)
+{
+    return msg->depth;
+}
+
+TEST(ControlSystem, depth)
+{
+    double test_depth = -2;
+    double overshoot_allowed  = 0.1;
+
+    rs::SubscriberAnalyzer<robosub::depth_stamped> analyzer;
+
+
+    analyzer.Init("depth", &get_depth_data);
```

Comment:
When passing this function as a pointer, you don't need to dereference it as stated in [this post](http://stackoverflow.com/questions/19417826/passing-function-pointer), though they also state it will still work this way.